### PR TITLE
Capacity: Has connections

### PR DESCRIPTION
**Pull request by ccailly**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 

image

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
From my tests your fix work nicely and the search option work now as expected (still l'll let @ccailly confirm).

However, it does not seems to fix the two broken tests (which are maybe unrelated from this search option issue ?)

**Comment by cconard96** [comment]
> However, it does not seems to fix the two broken tests (which are maybe unrelated from this search option issue ?)

For that, it looks like the join for the printer table is missing completely. I don't think there is an issue in the search engine with joins on their own, but maybe.

**Comment by ccailly** [comment]
> > I'm having a little trouble with the Search options. I had to add specific arguments to correctly handle the column names of the polymorphic `Connection_Item` class. However, it seems that the condition isn't working. Indeed, if I retrieve the final query executed by the code, the condition isn't present.
> 
> I had the same headache yesterday with the "multiple asset groups" PR. Seems like I headache I caused while moving the search code to separate files and cleaning it up. Can you confirm this fix resolves the issue for you too? I'll open a new PR with the fix if so.
> 
> https://github.com/glpi-project/glpi/pull/16760/files#diff-7a48f737d70d624c2b60cfaf63cfb8635d5a4f0b0748d97451d69f27a0acfddeL2484-R2485

I tried your code and got the same result as @AdrienClairembault. However, many others tests break after modification.



> > However, it does not seems to fix the two broken tests (which are maybe unrelated from this search option issue ?)
>
> For that, it looks like the join for the printer table is missing completely. I don't think there is an issue in the search engine with joins on their own, but maybe.

There are probably errors elsewhere, but in any case the number returned by the search option was not the right one and returned the total number of connections for the item. The condition is not present in the final SQL query.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]
There are two references to the old itemtype in the API documentation:

image

Can you update them ?

We also need to indicate that the class was renamed in the changelog.

@cedric-anne do we need to handle backward compatibility for the low level API ?

File: install/migrations/update_10.0.x_to_11.0.0/connections.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+$default_charset = DBConnection::getDefaultCharset();
+$default_collation = DBConnection::getDefaultCollation();
+$default_key_sign = DBConnection::getDefaultPrimaryKeySignOption();
+
+if (!$DB->tableExists('glpi_connections_items')) {
+    $query = <<<SQL
+        CREATE TABLE `glpi_connections_items` (
+          `id` int {$default_key_sign} NOT NULL AUTO_INCREMENT,
+          `items_id_item` int {$default_key_sign} NOT NULL DEFAULT '0' COMMENT 'RELATION to various table, according to itemtype_item (ID)',
+          `itemtype_peripheral` varchar(255) NOT NULL,
```
Comment (AdrienClairembault) [inline install/migrations/update_10.0.x_to_11.0.0/connections.php:51]:
```suggestion
          `itemtype_item` varchar(255) NOT NULL,
```

**Comment by cedric-anne** [comment]
> @cedric-anne do we need to handle backward compatibility for the low level API ?

As we have done it so far, yes, I guess.

**Comment by cedric-anne** [comment]
`Connection_Item` may be ambiguous. Indeed, one may think that it is related to network connections.
The relation described here is between an asset and a peripheral. Also, when we will implement the `IsAPeripheral` capacity, any generic asset would be able to be described as a peripheral. Maybe `Asset_PeripheralAsset` would be a more appropriate class name.
@orthagh @trasher , what is your opinion ?

**Comment by orthagh** [comment]
Ok for me

**Review by cedric-anne — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]
Functional test is OK for me.

Code look fine too, the main points I raised before seems to be fixed (but I can't check every new changes in details as the commits have been squashed).

**Pull request merged**